contrib/bump-version.sh

```diff
@@ -4,24 +4,35 @@
 # the CURRENT_VERSION to the TARGET_VERSION. This is meant as a helper - but
 # probably should still double-check the changes are correct
 
-if [ $# -ne 1 ]; then
+if [ $# -ne 1 ] || [ `expr match $1 ".*_.*"` == 0 ]; then
     echo "USAGE: $0 <target-version>"
-    echo "  example: $0 'v1.3.4_coreos.0'"
+    echo "  example: $0 'v1.3.5_coreos.1'"
     exit 1
 fi
 
-CURRENT_VERSION=${CURRENT_VERSION:-"v1.3.4_coreos.0"}
+CURRENT_VERSION=${CURRENT_VERSION:-"v1.3.5_coreos.1"}
 TARGET_VERSION=${1}
 
+CURRENT_INDEX=`expr index $CURRENT_VERSION _`
+TARGET_INDEX=`expr index $TARGET_VERSION _`
+
+CURRENT_VERSION_BASE=${CURRENT_VERSION:0:$CURRENT_INDEX -1}
+TARGET_VERSION_BASE=${TARGET_VERSION:0:$TARGET_INDEX -1}
+
+CURRENT_VERSION_BUILD=${CURRENT_VERSION:$CURRENT_INDEX}
+TARGET_VERSION_BUILD=${TARGET_VERSION:$TARGET_INDEX}
```

Comment:
If you really want to add something like this, maybe just 2 passes:

1 pass for full version string: `vx.y.z_coreos.v`
1 pass for upstream version: `vx.y.z`